### DISCUSSION TITLE
[New Feature] Proper INI inheritance

### DISCUSSION THREAD

**Discussion by Metadorius** [Phobos-developers/Phobos#943] [Ideas and feedback]
### Description:
The current system for inheritance in Ares is buggy, limited and doesn't comply with INI syntax. The proposed feature should work with any INI-defined types and allow multiple inheritance.

### Possible Implementation:
Two possible ways of implementation:
1. Parse `Inherits` tag when parsing any section (allows inheriting anything from anything without types even knowing inheritance was made) - parser level implementation;
2. Parse `Inherits` tag on type parse (limited inheritance only on selected types; allows inheritance in scenario rules).

Example INI code:
```ini
[TypeA]
A=1

[TypeB]
A=2
B=2

[TypeC] 
Inherits=TypeA,TypeB
C=3
; results in A=2 B=2 C=3

[TypeD]
Inherits=TypeB,TypeA
B=4
D=4
; results in A=1 B=4 D=4
```

**Comment by TAK02** [comment]
Maybe add `Inherits.Update=boolean ;def=no` too?
Basically, if the type inherited from gets any additions/overrides (via map files or multiplayer mode INIs) they're applied to the inheritor too.

Example:
```ini
[TypeA]
...
[TypeB]
Inherits=TypeA
Inherits.Update=yes
FlagA=...
[TypeC]
Inherits=TypeA
Inherits.Update=no
FlagA=...
```
When [TypeA] is given, say, `TechLevel=-1` in a map file, that TechLevel is applied to both [TypeA] and [TypeB] (`*.Update=yes`), but not [TypeC] (`*.Update=no`).
